### ISSUE TITLE
join does not respect column order

### ISSUE THREAD

**Issue by jkleint** [BurntSushi/xsv#11]
And maybe that's intended, but it seems like it would be useful to (at least have the option to) say join on "fileA.col1 = fileB.col2 and fileA.col2 = fileB.col1".  Right now these both give the same results:
```
xsv join 1,2 a.txt 1,2 b.txt
xsv join 1,2 a.txt 2,1 b.txt 
```
I expected the second to match column 1 from a.txt to column 2 from b.txt.

By the way, thanks for a great (and very fast) tool.

**Comment by BurntSushi** [comment]
Column order *should* definitely be respected. That's a major bug if it isn't. I'll try to reproduce later.

For giggles, could you post your `a.txt` and `b.txt`? Just so I can make sure we're working from the same data.

**Comment by jkleint** [comment]
```
$ xsv --version
0.9.2

$ head a.txt b.txt
==> a.txt <==
1,2
3,4
5,6

==> b.txt <==
2,1
4,3
6,5

$ xsv join -n 1,2 a.txt 2,1 b.txt

$ xsv join -n 1,2 a.txt 2,1 a.txt
1,2,1,2
3,4,3,4
5,6,5,6

$ xsv join -n 1,2 a.txt 1,2 a.txt
1,2,1,2
3,4,3,4
5,6,5,6
```

Maybe I'm doing something wrong? The first join seems like it should produce all rows.  The second, a self-join on different columns, seems like it should produce none, but it gives the same results as the last, which is what I expect.

**Comment by BurntSushi** [comment]
Very nice catch. You are absolutely right. I've fixed this and added your test case as a regression test. I uploaded the fix in a new release `0.9.3`. Thanks!

**Comment by jkleint** [comment]
Sweet, thanks; 0.9.3 works for me.
